File: VECTOAux/VectoAuxiliaries/Hvac/ISSMTechList.vb

Namespace Hvac

Public Interface ISSMTechList

 property TechLines As List(Of ITechListBenefitLine) 

 Sub Clear()
 Function    Add( item As ITechListBenefitLine, byref feedback as String) As Boolean
 Function Delete( item As ITechListBenefitLine, byref feedback as String) As Boolean
 Function Modify( originalItem As ITechListBenefitLine, modifiedItem As ITechListBenefitLine, byref feedback as String) As Boolean

  Sub SetSSMGeneralInputs( genInputs As ISSMGenInputs)

 ReadOnly Property HValueVariation  As Double
 ReadOnly Property VHValueVariation As Double
 ReadOnly Property VVValueVariation As Double
 ReadOnly Property VCValueVariation As Double
 ReadOnly Property CValueVariation  As Double

 ReadOnly Property HValueVariationKW  As Double
 ReadOnly Property VHValueVariationKW As Double
 ReadOnly Property VVValueVariationKW As Double
 ReadOnly Property VCValueVariationKW As Double
 ReadOnly Property CValueVariationKW  As Double


 Function Initialise() As Boolean
 Function Initialise( filePath As String ) As Boolean




End Interface



End Namespace

File: VectoCommon/VectoCommon/BusAuxiliaries/ISSMTechList.cs
using System.Collections.Generic;

namespace TUGraz.VectoCommon.BusAuxiliaries
{
	public interface ISSMTechList
	{
		IReadOnlyList<ISSMTechnology> TechLines { get; set; }

		//void Clear();
		//ITechListBenefitLine Find(string category, string benefitName);
		//bool Add(ITechListBenefitLine item, ref string feedback);
		//bool Delete(ITechListBenefitLine item, ref string feedback);
		//bool Modify(ITechListBenefitLine originalItem, ITechListBenefitLine modifiedItem, ref string feedback);

		//void SetSSMGeneralInputs(ISSMInputs genInputs);

		double HValueVariation { get; }
		double VHValueVariation { get; }
		double VVValueVariation { get; }
		double VCValueVariation { get; }
		double CValueVariation { get; }

		//double HValueVariationKW { get; }
		//double VHValueVariationKW { get; }
		//double VVValueVariationKW { get; }
		//double VCValueVariationKW { get; }
		//double CValueVariationKW { get; }


		//bool Initialise();
		//bool Initialise(string filePath);
	}
}
